packages/openspp/src/Adaptor.js

```diff
@@ -0,0 +1,705 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+  expandReferences,
+} from "@openfn/language-common";
+
+import pkg from "odoo";
+const { Odoo } = pkg;
+
+var sppConnector = null;
+
+
+export class Log {
+  static info(message) {
+    return console.info(`ℹ `, message);
+  }
+
+  static success(message) {
+    return console.info(`✓ Success at ${new Date()}:\n∟`, message);
+  }
+
+  static warn(message) {
+    return console.warn(`⚠ Warning at ${new Date()}:\n∟`, message);
+  }
+
+  static error(message) {
+    return console.error(`✗ Error at ${new Date()}:\n∟`, message);
+  }
+}
+
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create("foo"),
+ *   delete("bar")
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  sppConnector = null;
+
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(...operations)({ ...initialState, ...state });
+  };
+}
+
+/**
+ * Logs in to OpenSpp, gets a session token.
+ * @example
+ *  login(state)
+ * @private
+ * @param {State} state - Runtime state.
+ * @returns {State}
+ */
+function login(state) {
```

Comment:
This function looks asynchronous to me - which means when it returns, the login may not have completed and subsequent calls to the connector may fail.

I would recommend that this function returns a Promise, which is resolved inside the connect callback. I'd be happy to show you what I mean if that doesn't make sense or you're unfamiliar with the style!

Reply:
@josephjclark hi, I changed the style from callback to async await, can you re-review it again?